shell/browser/api/electron_api_utility_process.cc

```diff
@@ -0,0 +1,611 @@
+  transferable_message.ports = MessagePort::DisentanglePorts(
+      args->isolate(), wrapped_ports, &threw_exception);
+  if (threw_exception)
+    return;
+
+  node_service_remote_->ReceivePostMessage(std::move(transferable_message));
+}
+
+bool UtilityProcessWrapper::Kill() const {
+  if (pid_ == base::kNullProcessId)
+    return 0;
+  base::Process process = base::Process::Open(pid_);
+  bool result = process.Terminate(content::RESULT_CODE_NORMAL_EXIT, false);
+  // Refs https://bugs.chromium.org/p/chromium/issues/detail?id=818244
+  // Currently utility process is not sandboxed which
+  // means Zygote is not used on linux, refs
+  // content::UtilitySandboxedProcessLauncherDelegate::GetZygote.
+  // If sandbox feature is enabled for the utility process, then the
+  // process reap should be signaled through the zygote via
+  // content::ZygoteCommunication::EnsureProcessTerminated.
+  base::EnsureProcessTerminated(process.Duplicate());
+  return result;
+}
+
+v8::Local<v8::Value> UtilityProcessWrapper::GetOSProcessId(
+    v8::Isolate* isolate) const {
+  if (pid_ == base::kNullProcessId)
+    return v8::Undefined(isolate);
+  return gin::ConvertToV8(isolate, pid_);
+}
+
+void UtilityProcessWrapper::ReceivePostMessage(
+    blink::TransferableMessage message) {
+  v8::Isolate* isolate = JavascriptEnvironment::GetIsolate();
+  v8::HandleScope handle_scope(isolate);
+  v8::Local<v8::Value> message_value =
+      electron::DeserializeV8Value(isolate, message);
+  Emit("message", message_value);
+}
+
+void UtilityProcessWrapper::HandleMessage(ReaderType type,
+                                          std::vector<uint8_t> message) {
+  v8::Isolate* isolate = JavascriptEnvironment::GetIsolate();
+  v8::HandleScope handle_scope(isolate);
+  auto array_buffer = v8::ArrayBuffer::New(isolate, message.size());
+  auto backing_store = array_buffer->GetBackingStore();
+  memcpy(backing_store->Data(), message.data(), message.size());
+  if (type == ReaderType::STDOUT) {
+    Emit("stdout", array_buffer,
+         base::BindRepeating(&UtilityProcessWrapper::ResumeReading,
+                             weak_factory_.GetWeakPtr(), stdout_reader_.get()));
+  } else {
+    Emit("stderr", array_buffer,
+         base::BindRepeating(&UtilityProcessWrapper::ResumeReading,
+                             weak_factory_.GetWeakPtr(), stderr_reader_.get()));
+  }
+}
+
+void UtilityProcessWrapper::ResumeReading(PipeReaderBase* pipe_io) {
+  if (!pipe_io->is_shutting_down()) {
+    pipe_io->StartReadLoop();
+  }
+}
+
+void UtilityProcessWrapper::ShutdownReader(ReaderType type) {
+  if (type == ReaderType::STDOUT) {
+    PipeIOBase::Shutdown(std::move(stdout_reader_));
+  } else {
+    PipeIOBase::Shutdown(std::move(stderr_reader_));
+  }
+}
+
+// static
+gin::Handle<UtilityProcessWrapper> UtilityProcessWrapper::Create(
+    gin::Arguments* args) {
+  gin_helper::Dictionary opts;
+  if (!args->GetNext(&opts)) {
+    args->ThrowTypeError("Expected a dictionary");
```

Comment:
```suggestion
    args->ThrowTypeError("Options must be an object");
```

JS doesn't really use the term 'dictionary'

Reply:
https://github.com/electron/electron/pull/34980/commits/4a3a57985a314f72bf2afa0b8280d812aeec9c6a